Created 1011.capacity-to-ship-packages-within-d-days-en.md & Updated Readme.en.md (#360)
azl397985856/leetcode: problems/1011.capacity-to-ship-packages-within-d-days-en.md

Return the least weight capacity of the ship that will result in all the packages on the conveyor belt being shipped within D days.

**Example 1:**

```
Input: weights = [1,2,3,4,5,6,7,8,9,10], D = 5
Output: 15
Explanation: 
A ship capacity of 15 is the minimum to ship all the packages in 5 days like this:
1st day: 1, 2, 3, 4, 5
2nd day: 6, 7
3rd day: 8
4th day: 9
5th day: 10

Note that the cargo must be shipped in the order given, so using a ship of capacity 14 and splitting the packages into parts like (2, 3, 4, 5), (1, 6, 7), (8), (9), (10) is not allowed. 
```

**Example 2:**

```
Input: weights = [3,2,2,4,1,4], D = 3
Output: 6
Explanation: 
A ship capacity of 6 is the minimum to ship all the packages in 3 days like this:
1st day: 3, 2
2nd day: 2, 4
3rd day: 1, 4
```

**Example 3:**

```
Input: weights = [1,2,3,1,1], D = 4
Output: 3
Explanation: 
1st day: 1
2nd day: 2
3rd day: 3
4th day: 1, 1
```

 **Note:**

1. `1 <= D <= weights.length <= 50000`
2. `1 <= weights[i] <= 500`

## Solution

The problem is same as [**LeetCode 875 koko-eating-bananas**] (https://github.com/azl397985856/leetcode/blob/master/problems/875.koko-eating-bananas-en.md) practically.

> It is easy to solve the problem of this type if you can discover its essence.

Reviewer: ```suggestion
It is easy to solve this kind of problems  if you take a closer look into it.

```

The essence is to search a given number in finite discrete data like [ 1,2,3,4, ... , total ].  

> We could find the cargo that can be shipped in D days rather than look for the target directly.

Reviewer: ```suggestion
However, We should find the cargo that can be shipped in D days rather than look for the target directly.

```

Consider the following questions:

- Can it be shipped if the capacity is 1?
- Can it be shipped if the capacity is 2?
- Can it be shipped if the capacity is 3?
- ...
- Can it be shipped if the capacity is total ? ( Yeap we can,  D is greater than or equal to 1)

During the process, we  directly `return`  if the answer is *yes*.

If the answer is *no*,  just keep asking.

> The question is a typical binary search problem, a little bit difference is the judgement condition:

Reviewer: ```suggestion
This is a typical binary search problem, the only difference is the judgement condition:

```
